### PR TITLE
ARROW-3172: [Rust] Update documentation for datatypes.rs

### PR DESCRIPTION
**Pull request by paddyhoran**

### CODE REVIEW

File: rust/src/datatypes.rs [hunk 1]
```diff
@@ -197,19 +222,22 @@ impl Field {
         }
     }
 
+    /// Returns a immutable reference to the `Field`'s name
```
Comment (kszucs) [inline rust/src/datatypes.rs]: `an` immutable

File: rust/src/datatypes.rs [hunk 2]
```diff
@@ -197,19 +222,22 @@ impl Field {
...
     pub fn name(&self) -> &String {
         &self.name
     }
 
+    /// Returns a immutable reference to the `Field`'s  data-type
```
Comment (kszucs) [inline rust/src/datatypes.rs]: `an`

File: rust/src/datatypes.rs [hunk 3]
```diff
@@ -15,14 +15,32 @@
 // specific language governing permissions and limitations
 // under the License.
 
+/*!
```
Comment (kszucs) [inline rust/src/datatypes.rs]: Shouldn't this use `//!` style comments instead?

Comment (paddyhoran) [reply rust/src/datatypes.rs]: I believe that surrounding with `/*!` and `*/` is equivalent to starting each line with `//!`.  I'm finding it hard to confirm this but I noticed this approach [here](https://github.com/BurntSushi/rust-csv/blob/master/src/lib.rs)

I guess we should stick with `//!` as it seems to be most used approach.

**Review by andygrove — APPROVED** [review]
LGTM

**Comment by paddyhoran** [comment]
@kszucs @andygrove I'm not sure why the windows build failed.  The previous commit only failed the non-windows build because thats where we run the `fmt` tool.

I just ran everything locally on windows and it's all good, I think Appveyor just timed out or something.  I think this should be good to merge (not sure how to re-trigger CI without making changes to confirm my theory), what do you think?

**Review by kszucs — APPROVED** [review]

**Comment by kszucs** [comment]
It was a temporary network issue. Thanks @paddyhoran!

**Pull request closed without merging**